core/interop/tests/module.rs

```diff
@@ -0,0 +1,112 @@
+//! Test for the class proc-macro.
+#![allow(unused_crate_dependencies)]
+
+use boa_engine::{js_string, Context, JsString, Module, Source};
+use boa_macros::{boa_class, boa_module, Finalize, JsData, Trace};
+use std::rc::Rc;
+
+#[derive(Clone, Trace, Finalize, JsData)]
+enum AnimalType {
+    Cat,
+    Dog,
+    Other,
+}
+
+#[derive(Clone, Trace, Finalize, JsData)]
+struct Animal {
+    ty: AnimalType,
+    age: i32,
+}
+
+#[boa_class]
+#[boa(rename = "camelCase")]
```

Comment:
I don't think these should be independent attributes. It doesn't make sense to have `#[boa(rename = "camelCase")]` without a corresponding `#[boa_class]` or `#[boa_module]`. I would make it so that you're required to provide the rename schema as an option for the `boa_class` or `boa_module` attributes:

`#[boa_class(rename = "camelCase")]`